Make release-process.md up to date (#300)
kyma-project/community: guidelines/releases/release-process.md

> Define release jobs on the `master` branch in the `test-infra` repository. To ensure every job name is unique, prefix it with `pre-rel{versionNumber}`. Remember to provide the version number without any periods. To learn how to define a release job for a component, read the following [document][2]. To find all jobs for the 0.9 release, look for job names with the `pre-rel09` prefix.

Reviewer: Move the last sentence as an example (one sentence earlier). Also: finish the sentence in the previous paragraph.

1. Navigate to the `test-infra` repository.

> 2. Define release jobs in the `prow/jobs/test-infra` directory in the following files:

Reviewer: ```suggestion
2. Define release jobs in the `prow/jobs/test-infra` directory in the `watch-pods.yaml` file.
```

> 	- `watch-pods.yaml`

Reviewer: Delete this line (one-item lists are not acceptable).

3. Define release jobs in the `prow/jobs/kyma` directory in the following files:

>    1. every `.yaml` in `components` 

Reviewer: Please revert this numbered list into bullet points.

   2. every `.yaml` in `tests`

>    3. every `.yaml` in `tools/` \*\*except for\*\*:

Reviewer: ```suggestion
   3. every `.yaml` in `tools/` except for the `tools/failery/failery.yaml` file
```

> 	 - `tools/failery/failery.yaml`

Reviewer: Delete this line (no for one-item lists).

   4. `kyma-integration.yaml`
   5. `kyma-artifacts.yaml`
   6. `kyma-installer.yaml`
   7. `kyma-github-release.yaml`
   8. `kyma-release-candidate.yaml`

> 		> **Note:** Remember to modify the `presets` array. This is an example of the 0.9 release `preset`:

Reviewer: ```suggestion
		> **NOTE:** Remember to modify the `presets` array. This is an example of the 0.9 release `preset`:
```

		```

1. Ensure that tests for the release jobs exist. Release tests usually iterate through all release versions and run tests for them.
See the `TestBucReleases` test defined in `development/tools/jobs/kyma/service_binding_usage_controller_test.go` as a reference.

1. Update the `GetAllKymaReleaseBranches()` function
defined in the `development/tools/jobs/tester/tester.go` file under the `test-infra` repository.

1. Define branch protection rules for the release branch in the `prow/config.yaml` file.

> 

Reviewer: Is there any need for such space?
